clusterpools/apply.sh

```diff
@@ -510,14 +594,88 @@ if [[ "$CLUSTERPOOL_NAME" == "" ]]; then
     if [[ "$INPUT_CLUSTERPOOL_NAME" == "" ]]; then
         CLUSTERPOOL_NAME="$GENERATED_CLUSTERPOOL_NAME-$(echo $PLATFORM | tr '[:upper:]' '[:lower:]')-$(echo $CLUSTERIMAGESET_NAME | sed 's/\.//g')"
     else
+        CUSTOM_CLUSTERPOOL_NAME="true"
         CLUSTERPOOL_NAME="$INPUT_CLUSTERPOOL_NAME"
     fi
 fi
 printf "${GREEN}* Using Name: $CLUSTERPOOL_NAME${CLEAR}\n"
 
 
+#-----OPTIONALLY CONFIGURE A CUSTOM INSTALL CONFIG SECRET-----#
+if [[ "$YQ_INSTALLED" != "false" ]]; then
+    if [[ "$CUSTOM_CLUSTERPOOL_NAME" == "true" ]]; then
+        CLUSTERPOOL_INSTALL_CONFIG_SECRET_NAME="${CLUSTERPOOL_NAME}-${RANDOM_IDENTIFIER}-install-config-template"
+    else
+        CLUSTERPOOL_INSTALL_CONFIG_SECRET_NAME="${CLUSTERPOOL_NAME}-install-config-template"
+    fi
+    if [[ "$CLUSTERPOOL_INSTALL_CONFIG_FILE" == "" ]]; then
+        printf "${BLUE}- note: to skip this step in the future and create a new install-config secret from a file automatically, export CLUSTERPOOL_INSTALL_CONFIG_FILE${CLEAR}\n"
+        if [[ "$CLUSTERPOOL_INSTALL_CONFIG_SECRET" == "" ]]; then
+            printf "${BLUE}- note: to skip this step in the future and use a pre-existing install-config secret, export CLUSTERPOOL_INSTALL_CONFIG_SECRET${CLEAR}\n"
+            printf  "${BLUE}- note: if you choose 'Y', you must have list permissions for secrets on the namespace '$CLUSTERPOOL_TARGET_NAMESPACE'.${CLEAR}\n"
+            printf "${YELLOW}Do you want to create a custom install-config.yaml for your clusterpool clusters? (Y/N) ${CLEAR}"
+            read selection
+            if [[ "$selection" == "Y" || "$selection" == "y" ]]; then
+                # Prompt the user to select an install-config secret
+                secrets=$(oc get secret -n $CLUSTERPOOL_TARGET_NAMESPACE 2> /dev/null)
+                if [[ "$?" != "0" ]]; then
+                    printf "${BLUE}- It looks like you don't have access to list secrets in $CLUSTERPOOL_TARGET_NAMESPACE, so we won't try to create or set one (our query errored).${CLEAR}\n"
+                    printf "${YELLOW}Do you wish to continue without setting a custom install-config.yaml?${CLEAR}\n"
+                    read selection
+                    if [[ "$selection" != "Y" && "$selection" != "y" ]]; then
+                        printf "${BLUE}Exiting.\n${CLEAR}"
+                        exit 1
+                    fi
```

Comment:
Should this be "N"? We don't want to exit if they want to continue, do we?

Reply:
Ahh - the logic here is "do you wish to continue" and if the answer is _not_ "Y" or "y" we bail

Reply:
Oh yeah--that makes sense. Didn't notice the negation 🙂